lib/main.coffee

```diff
@@ -61,6 +61,13 @@ Notifications =
     @addNotificationsLogSubscriptions() if @notificationsLog?
     @subscriptions.add atom.workspace.addOpener (uri) => @createLog() if uri is NotificationsLog::getURI()
     @subscriptions.add atom.commands.add 'atom-workspace', 'notifications:toggle-log', -> atom.workspace.toggle(NotificationsLog::getURI())
+    @subscriptions.add atom.commands.add 'atom-workspace', 'notifications:clear-log', =>
+      for notification in atom.notifications.getNotifications()
+        notification.options.dismissable = true
+        notification.dismissed = false
```

Comment:
Why is this needed?  Under what circumstances will there be an already-dismissed notification in the list that we need to dismiss again?

Reply:
non-dimissable notifications (ones that dismiss automatically) have [`.dismissed` = true](https://github.com/atom/atom/blob/a3108623de406e6f162c37b2dffd0872ab39e276/src/notification.js#L13) when they are first shown. If someone clears the log we want those notifications to be dismissed right away not wait for the timer to dismiss them automatically.